app/server.js

```diff
@@ -4,28 +4,195 @@ const jwt = require('jsonwebtoken');
 const swaggerUi = require('swagger-ui-express');
 const YAML = require('yamljs');
 const swaggerDocument = YAML.load('./swagger.yaml'); // Replace './swagger.yaml' with the path to your Swagger file
+const cors = require('cors');
+
 const app = express();
 
+const PRIVATE_KEY = "ILoveSunglasses";
+
+app.use(cors({
+	origin: 'http://localhost:3000',
+	methods: ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'],
+	allowedHeaders: ['Authorization', 'Content-Type'],
+	credentials: true
+}));
+
 app.use(bodyParser.json());
 
 // Importing the data from JSON files
 const users = require('../initial-data/users.json');
 const brands = require('../initial-data/brands.json');
 const products = require('../initial-data/products.json');
 
+
+const verifyToken = (req, res, next) => {
+	const authHeader = req.headers.authorization;
+	if (!authHeader) {
+		return res.status(401).json({ error: 'No token provided' });
+	}
+	const token = authHeader.split(' ')[1];
+	try {
+		const decoded = jwt.verify(token, PRIVATE_KEY);
+		req.username = decoded.username;
+
+		const payload = jwt.decode(token, { complete: true }).payload;
+		const currentExp = payload.exp;
+
+		if (currentExp) {
+			const updatedExp = Math.floor(Date.now() / 1000) + 3600;
+			payload.exp = updatedExp;
+
+			const updatedToken = jwt.sign(payload, PRIVATE_KEY);
+			res.locals.newToken = updatedToken;
+		}
+
+		next();
+	} catch (error) {
+		res.status(401).json({ error: 'Token invalid or expired' });
+	}
+};
+
 // Error handling
 app.use((err, req, res, next) => {
 	console.error(err.stack);
 	res.status(500).send('Something broke!');
 });
 
+app.get('/brands', (req, res) => {
+	res.status(200).json(brands);
+});
+
+app.get('/products', (req, res) => {
+	res.status(200).json(products);
+});
+
+app.post('/login', (req, res) => {
+	const { username, password } = req.body;
+	const user = users.find(user => user.login.username === username && user.login.password === password);
+	if (!user) {
+		return res.status(401).json({ error: 'Invalid login info' });
+	}
+	const token = jwt.sign({ username: username }, PRIVATE_KEY, { expiresIn: '1h' });
+	res.locals.newToken = token;
+	res.status(200).json({
+		cart: user.cart,
+		token: res.locals.newToken
+	});
+});
+
+app.get('/brands/:id/products', (req, res) => {
+	const brandId = req.params.id;
+	const targetBrand = brands.filter(brand => brand.id === brandId);
+	if (targetBrand.length === 0) {
+		return res.status(404).json({ error: 'Invalid brand ID' });
```

Comment:
this message is too technical, imagine you get this as web page user. Something better would be: 
```suggestion
		return res.status(404).json({ error: 'Brand not found' });
```